.github/workflows/pr-go-workspace-check.yml

```diff
@@ -0,0 +1,39 @@
+name: "Go Workspace Check"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**/go.work'
+      - '**/go.work.sum'
+      - '**/go.mod'
+      - '**/go.sum'
+
+jobs:
+  check:
+    name: Go Workspace Check
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set go version
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.21.8'
+
+    - name: Build
+      run: |
+        go mod tidy && make build-go
```

Comment:
what about `go work sync`?

Reply:
yeah, i have been trying to figure that out. it seems like there are multiple situations (tidy, sync, build) that will cause an update to go.work.sum, but they aren't deterministic 🙃 

for sync, i think it depends if we want to force dependency updates on all sub-modules. thoughts?

Reply:
I would go for sync. With that way we will make sure all is up-to-date. But this might be also scary for the people who would see too many changes.

Reply:
yeah, if we don't have an automated way of updating other go.mods, those are doomed to get outdated